### PR TITLE
ConnectionData bindings

### PR DESCRIPTION
**Pull request by Aksem**

### CODE REVIEW

File: tests/test_connectiondata.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    connection_data.setNeedsToken(request_name)
+    assert connection_data.needsToken(request_name) == True
+    assert connection_data.needsToken('false_request_name') == False
+
+
+def test_connectiondata_setlastevent(connection_data):
+    request_name = 'random_request_name'
+
+    connection_data.setNeedsToken(request_name)
```
Comment (KitsuneRal) [inline tests/test_connectiondata.py:67]: Seems like copy-pasta.

**Review by KitsuneRal — COMMENTED** [review]
Overall looks reasonable, except that probably it's better to use Python accessors, as discussed in the chat; and one copy-paste leftover.

File: tests/test_connectiondata.py (hunk 1 above)
Comment (Aksem) [reply tests/test_connectiondata.py]: Yes, it was. Fixed.

**Comment by Aksem** [comment]
> Overall looks reasonable, except that probably it's better to use Python accessors, as discussed in the chat; and one copy-paste leftover.

Changed API so that it now conforms PEP8(snake case) and also Python properties are used.

**Review by KitsuneRal — APPROVED** [review]
Looks great :)

**Review by KitsuneRal — APPROVED** [review]

**Pull request merged**